### PR TITLE
pr-1

### PR DESCRIPTION
**Pull request by bohdan2704**
added layers to app

### CODE REVIEW

File: src/main/java/com/example/bookshop/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String title;
+
+    private String author;
+
+    private String isbn;
+
+    private BigDecimal price;
```
Comment (Rommelua) [inline src/main/java/com/example/bookshop/model/Book.java:22]: add constraints

File: src/main/java/com/example/bookshop/exception/DataProcessingException.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.example.bookshop.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String message) {
+        super(message);
+    }
+
```
Comment (Rommelua) [inline src/main/java/com/example/bookshop/exception/DataProcessingException.java:7]:
```suggestion
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookshop/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    @NotBlank
+    private String isbn;
+
+    @NotNull
+    private BigDecimal price;
+
+    private String description;
+
+    private String coverImage;
```
Comment (Vasul-Pankiv) [inline src/main/java/com/example/bookshop/model/Book.java:38]: remove empty lines

File: src/main/java/com/example/bookshop/repository/BookRepository.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.example.bookshop.repository;
+
+import com.example.bookshop.model.Book;
+import java.util.List;
+
+public interface BookRepository {
```
Comment (Vasul-Pankiv) [inline src/main/java/com/example/bookshop/repository/BookRepository.java:6]: try to use JpaRepository, so you will no longer need to create an implementation

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookshop/exception/DataProcessingException.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.example.bookshop.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String message, Throwable e) {
+        super(message);
```
Comment (kerrrusha) [inline src/main/java/com/example/bookshop/exception/DataProcessingException.java:5]: you dont used this parameter
```suggestion
        super(message, e);
```

**Review by kerrrusha — APPROVED** [review]

**Pull request merged**